---
title: "Discipline Project 2"
author: Ricky Cai (440242475)
geometry: margin=2cm
output:
  pdf_document:
    pandoc_args: [
      "-V", "classoption=twocolumn"
      ]
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    theme: united
    code_folding: hide
---

# Topic

<<<<<<< HEAD
As the technology for sequencing genes becomes more readily available, more powerful and cheaper, the information about genes that is captured becomes more detailed. This becomes a problem when using machine learning techniques to analyse the high dimensional data. This problem is known as the "big p, little n" problem where p refers to the number of measurements and n refers to the subjects. Consequences of high dimension data include multiple hypothesis testing and high computation time. In this report, we analyse the GSE107509 gene expression data set that contains 659 samples and over 54,000 genes. This data set was chosen for its high number of samples leading to a higher diversity of samples.
=======
As the technology for sequencing genes becomes more readily available, more powerful and cheaper, the information about genes that is captured becomes more detailed. This becomes a problem when using machine learning techniques to analyse the high dimensional data. This problem is known as the "big p, little n" problem where n refers to the subjects and p refers to the number of measurements. Consequences of high dimension data include multiple hypothesis testing and high computation time. In this report, we analyse the GSE107509 gene expression data set that contains 659 samples and over 54,000 genes. This data set was chosen since due tot he high number of samples leading to a higher diversity of samples.
>>>>>>> parent of 2d6ec1a... Shiny app complete, finalising report and presentation

Dimension reduction techniques such as Principal Component Analysis (PCA) is a way to transform the high dimension data into low dimension data while attempting to preserve the original information as much as possible. PCA is used to help alleviate the problems associated with "big p, little n".

The topic that is addressed in this report analyses how dimension reduction techniques such as PCA affect the accuracy of random forest models. An accompanying Shiny App was developed to provide a way to interactively visualise the effect of how PCA (Principal Component Analysis) affects the accuracy of a random forest model on gene expression. 


# Approach

The approach to answering the question uses Principal Component Analysis to reduce the number of dimensions in the data, Random Forests to predict the outcome and cross validation to predict the accuracy of the models. The approach includes preselecting a number of principal components since it was not feasible to run in real time. Random Forest models were trained using the reduced number of principal components and cross validation was used with boxplots to show the effect of how the number of principal components affects the accuracy of the Random Forest model.

# Shortcomings

A shortcoming to the data analytics was the size of the data which had an effect on how the data was stored on Github (There is a 100MB limit on Github). To host the Shiny application on Github, the gene expression data was split into several parts and recombined in the Shiny app when it is run. 

<<<<<<< HEAD
Another shortcoming was the computation time for running PCA and Random Forest. To address this issue, the data was preprocessed and saved. Some of the data used in the Shiny app uses preprocessed data to reduce the time it takes to run models. This prevents the user in having full control over the parameters but a selection of preprocessed values is provided for comparison.
=======
Another shortcoming was the computation time for running PCA and Random Forest models. To address this issue, the data was preprocessed and saved. The data used in the Shiny app uses preprocessed data to reduce the time it takes to run models. This does not let the user have full control over the parameters but a selection of preprocessed values is provided for comparison.
>>>>>>> parent of 2d6ec1a... Shiny app complete, finalising report and presentation


# Graphics

There are 3 sections to the Shiny App. The sections include Calculator, PCA and Random Forest. 

The first section makes use of colour to visualise the top 12 genes (measured by the highest variance) and visualises how an individuals gene expression compares with the mean in the data set (Fig. 1). Red indicates the individual had a gene expression lower than the mean while green indicates the individual had a gene expression higher than the mean for that particular gene.

![Comparing individual gene expression with mean](images/Fig1.png)



The second section makes use of a screeplot of the PCA to show the cululative variance explained by the number of Principal Components selected (Fig. 2). The user can specify the number of Principal Components by a slider which dynamically shows the percentage of variance explained by the specified number of Principal Components.

![Screeplot of PCA](images/Fig2.png)

The third section makes use of boxplots to show how the number of Principal Components affects the accuracy of a Random Forest model (Fig. 3). The user can compare between 5, 10 25, 50, 75, 100, 150 and 200 components using a selection box which dynamically updates the boxplots.

![Boxplots showing effect of principal components on accuracy](images/Fig3.png)


# Shiny App

The Github link to the Shiny App is provided below: 
https://github.com/iRick92/DATA3888_Discipline_Project_2

<<<<<<< HEAD
[Github Link](https://github.com/iRick92/DATA3888_Discipline_Project_2)


The Shiny App can be accessed by the following lines of code in R (May take a few minutes to download):
=======
The Shiny App can be accessed through the running the following line of code in R:
>>>>>>> parent of 2d6ec1a... Shiny app complete, finalising report and presentation

```{r, eval = F}

library(shiny)
runGitHub("DATA3888_Discipline_Project_2", "iRick92")

```

